### DISCUSSION TITLE
Running two TeslaMate instances simultaneously

### DISCUSSION THREAD

**Discussion by eden881** [teslamate-org/teslamate#4012] [Q&A]
Hi!

I have TeslaMate running on my Raspberry Pi, collecting data. Occasionally, I also want to experiment with code changes to TeslaMate that I make on my PC.
Since I obviously don't have a separate car and account just for testing, I'd want to use my car for both instances.

Will TeslaMate run fine with two instances running? I can't see a reason why not since it just uses Tesla's API, but I figured maybe someone in the community already did this and can confirm.

**Comment by cwanja** [comment]
Disclosure - not running two instances of TeslaMate. 

But from what I can infer, the biggest risk you run is throttling from Tesla API. We have seen it where people are running two different third-party loggers and Tesla is really clamping down on this. Maybe a suggestion would be to run both only when you absolutely need to and make drives / charges that you do not care about and or are throw away data. And if the last point is 100% true, you could shut down your "production" instance and run only the development instance thus the throw away data is not captured in production.

It is a tricky setup given Tesla's clamp down on the API consumption.

**Reply by brianmay** [reply]
What Chris says. Running two instances on the same cars would effectively double the number of requests made. At a time when rumours are circulating about server rate limits and paid for access. Perhaps not good to do it right now.

TeslaMate itself - assuming you have different databases configured - should be fine with this. Even the legacy streaming API which TeslaMate uses should be fine.

**Reply by craigcabrey** [reply]
If you had two instances pointed at the same DB, would it generate duplicate logs or is there some sort of nonce used for each event?

**Reply by brianmay** [reply]
I think Teslamate never has been designed for that use case, so not sure what would happen.

**Comment by DrMichael** [comment]
Well, I am running always two instances. So, if one misses a drive or charge, I cann adjust for that.

No issues so far. (Ok, I am not driving a lot...)

**Reply by bahree** [reply]
I have three running and have been  for a long time now. No issues with it.
